src/com/palmergames/bukkit/towny/db/TownyFlatFileSource.java

```diff
@@ -349,28 +347,48 @@ public boolean loadResidentList() {
 		
 		TownyMessaging.sendDebugMsg("Loading Resident List");
 		String line = null;
+		List residents = new ArrayList<String>();
 
+		// Build up a list of residents from any existing legacy residents.txt files.
 		try (BufferedReader fin = new BufferedReader(new InputStreamReader(new FileInputStream(dataFolderPath + File.separator + "residents.txt"), StandardCharsets.UTF_8))) {
 			
-			while ((line = fin.readLine()) != null) {
-				if (!line.equals("")) {
-					try {
-						newResident(line);
-					} catch (AlreadyRegisteredException e) {
-						TownyMessaging.sendDebugMsg("Duplicate resident '" + line + "' found in residents.txt, ignoring.");
-						continue;
-					}
-				}
+			while ((line = fin.readLine()) != null && !line.equals(""))
+				residents.add(line);
+		} catch (Exception ignored) {
+			// No residents.txt any more.
+		}
+		File residentFolder = new File(dataFolderPath + File.separator + "residents");
+		File[] residentFiles = residentFolder.listFiles(new FileFilter() {
+			@Override
+			public boolean accept(File residentFolder) {
+				return residentFolder.getName().toLowerCase().endsWith(".txt"); 
+			}
+		});
```

Comment:
This can be condensed into a lambda:
```java
File[] residentFiles= residentFolder.listFiles((file) -> file.getName.toLowerCase().endsWith(".txt");
```